daffodil-runtime2/src/main/resources/examples/NestedUnion.c

```diff
@@ -379,7 +380,8 @@ data_unparseSelf(const data *instance, UState *ustate)
         bar_unparseSelf(&instance->bar, ustate);
         break;
     default:
-        ustate->error_msg = "node's _choice field was not initialized during unparsing";
+        ustate->error_msg =
+            "Unparse error: no match between choice dispatch key and any branch key";
```

Comment:
Another just abort() case.

Reply:
I think I'm wrong here also. This is unparse, I am not sure why we're looking at branch keys here though. Unparsing isn't symetric with parsing. It never needs to look at branch keys. The unparse problem is given an element in the infoset, what branch does that indicate exists. 
In Runtime1 it's called "choice branch resolver".

Reply:
If we implement a choice branch resolver in c-runtime2, I think we would call it in walkInfosetNode() because the resolver needs to peek at the next XML element and walkInfosetNode() reads the XML file's data into infoset member fields.  We probably should call a getChoice() function in unparseSelf() instead of accessing the _choice member field in case we want to remove the _choice member field from C structs.